proposals/networkservicemesh.adoc

```diff
@@ -0,0 +1,72 @@
+== Network Service Mesh Proposal
+
+*Name of project:* Network Service Mesh
+
+*Description*
+
+Network Service Mesh is a novel approach solving complicated L2/L3 use cases in Kubernetes that are tricky to address with the existing Kubernetes Network Model. Inspired by Istio, Network Service Mesh maps the concept of a service mesh to L2/L3 payloads.
+
+Network Service Mesh recognizes and respects that the existing K8s Networking provides excellent service for the common developer use cases, and has therefore adopted an orthogonal approach to the CNI that currently provides K8s Networking, leaving that path completely unaltered.
+It requires no changes of any kind to the CNI plugin being used, or to Kubernetes itself to function.  It runs as an additional layer of infrastructure on top of stock out of the box K8s.
+
+When Network Service Mesh is installed on a K8s cluster, a developer can simply specify in an annotation on their Pod the name of the Network Service they want, like "secure-intranet-connectivity" and Network Service Mesh will insure that is available to their Pods (not to the cluster as a whole) in addition to the standard K8s Networking provided by their CNI and in a manner that is completely non-conflicting with it.
+
+While much of the work being done in Network Service Mesh is focused on K8s support, it is designed to not be welded to K8s.  It has been architected to also bring its benefits to other environments, like VIMs, Physical Networks, or Legacy Servers.
+
+*Statement on alignment with CNCF mission*:
+
+Network Service Mesh is designed to allow any L2/L3 Network Service a workload may want to be dynamically available to it.   It is seeking to make networking itself Cloud-native by bringing acknowledging and working with the existing immutable infrastrcture in K8s to loosely couple whatever non-standard Networking needs a workload has with minimal toil.
+It adopts a networking paradigm of allowing the developer to name the Network Service they want, apply metadata in the form of labels, and be simply connected to it.  It forcing the 35 year old legacy networking concepts like "subnet", "route", "interface", or "ipam" on the developer.  Those have no place in a Cloud-native world.
```

Comment:
The last two sentences should be reworded.

Reply:
How about something like:
> NSM removes the need for a developer to use legacy networking concepts like "subnet", "route", "interface", or "ipam."  Instead, it adopts a cloud-native networking paradigm, allowing the developer to name the Network Service they want, apply metadata in the form of labels, and be simply connected to it.

Reply:
re: It is seeking to make networking itself Cloud-native by bringing acknowledging and working with the existing immutable infrastrcture in K8s to loosely couple whatever non-standard Networking needs a workload has with minimal toil.

1. "cloud native" can be lowercase and non hyphenated, per CNCF style guide https://github.com/cncf/foundation/blob/master/style-guide.md

2. "by bringing acknowledging and working" could be reworded for clarity

3. infrastrcture s/b infrastructure

4. Can "Networking" be lowercase?